orangewidget/utils/listview.py

```diff
@@ -68,6 +68,12 @@ def setModel(self, model: QAbstractItemModel) -> None:
         self.__pmodel.setSourceModel(model)
         self.__filter_reset()
         self.model().rowsInserted.connect(self.__model_rowInserted)
+        self.model().modelReset.connect(self.__on_modelReset)
+
+    def __on_modelReset(self):
+        self.__filter_reset()
+        self.__pmodel.setFilterFixedString("")
+        self.__pmodel.setFilterFixedString(self.__search.text())
```

Comment:
I don't like this. The goal is, obviously, to trigger some signal or some kind of update by changing the filter. Couldn't you trigger this directly? Apparently, setting the filter calls [`filter_changed`](https://github.com/radekp/qt/blob/master/src/gui/itemviews/qsortfilterproxymodel.cpp#L965), which you can trigger by calling `self.__pmodel.invalidateFilter()`. I tried this, but 

```python
    def __on_modelReset(self):
        self.__filter_reset()
        self.__pmodel.setFilterFixedString(self.__search.text())
        self.__pmodel.invalidateFilter()
```

doesn't work. Could you explain (perhaps next Friday, if we don't meet earlier), which side effect are you triggering by setting the filter to `""` and back again?